vm/devices/storage/disk_ramdisk/src/lib.rs

```diff
@@ -254,20 +241,52 @@ impl Unmap for RamDisk {
         Ok(())
     }
 
+    fn unmap_behavior(&self) -> UnmapBehavior {
+        // This layer zeroes if the lower layer is zero, but otherwise does
+        // nothing, so we must report unspecified.
+        UnmapBehavior::Unspecified
+    }
+
     fn optimal_unmap_sectors(&self) -> u32 {
         1
     }
 }
 
+impl WriteNoOverwrite for RamLayer {
+    async fn write_no_overwrite(
+        &self,
+        buffers: &RequestBuffers<'_>,
+        sector: u64,
+    ) -> Result<(), DiskError> {
+        self.write_maybe_overwrite(buffers, sector, false)
+    }
+}
+
+/// Create a RAM disk of `size` bytes.
+///
+/// This is a convenience function for creating a layered disk with a single RAM
+/// layer. It is useful since non-layered RAM disks are used all over the place,
+/// especially in tests.
+pub fn ram_disk(size: u64, read_only: bool) -> anyhow::Result<LayeredDisk> {
```

Comment:
aesthetically, I'd prefer to model this (and possibly also `RamDiffDisk`) as their structs (e.g: `struct RamDisk(LayeredDisk)`) with `new` ctors.

the downside, of course, is that you have to manually plumb through the `impl DiskIo`, which kinda stinks?

the freestanding function is just kinda... ehh.

but this is all minor stuff, so ¯\\\_(ツ)_/¯

Reply:
RAM disk is special enough since it gets used in tons of tests.

I actually modified this slightly to be even more convenient for tests, by having it wrap in a `Disk` as well. Which in some ways is better, because the `LayeredDisk` thing doesn't leak through.